### PR TITLE
Alerting docs: Specify Grafana alert rule evaluation

### PR DESCRIPTION
**Pull request by ppcano**
Discussed in Slack in reference to [Rule evaluation update](https://grafana.com/docs/grafana/latest/whatsnew/whats-new-in-v10-4/#rule-evaluation-spread-over-the-entire-evaluation-interval).

### CODE REVIEW

File: docs/sources/alerting/fundamentals/alert-rules/_index.md [hunk 1]
```diff
@@ -120,6 +120,5 @@ When choosing which alert rule type to use, consider the following comparison be
...
 | Use images in alert notifications                                              | Yes                                                                                                                          | No                                                                                                                                                      |
 | Organization                                                                   | Organize and manage access with folders                                                                                      | Use namespaces                                                                                                                                          |
-| Evaluation of alert rules in the same evaluation group                         | Concurrent evaluation                                                                                                        | Sequential evaluation                                                                                                                                   |
```
Comment (ppcano) [inline docs/sources/alerting/fundamentals/alert-rules/_index.md:123]: Now, there is no major difference in the evaluation engine between both alert rule types, so better to not display this in this comparison table.

File: docs/sources/alerting/fundamentals/alert-rule-evaluation/_index.md [hunk 2]
```diff
@@ -39,7 +39,7 @@ Each evaluation group contains an **evaluation interval** that determines how fr
 
 Alert rules in different groups can be evaluated simultaneously.
 
-**Grafana-managed** alert rules within the same group are evaluated simultaneously. However, **data-source managed** alert rules within the same group are evaluated one after the other—this is necessary to ensure that recording rules are evaluated before alert rules.
```
Comment (brendamuir) [inline docs/sources/alerting/fundamentals/alert-rule-evaluation/_index.md:42]: did konrad /dev confirm the behaviour?

Comment (gillesdemey) [reply docs/sources/alerting/fundamentals/alert-rule-evaluation/_index.md:42]: Yes, this seems accurate to me :)

File: docs/sources/alerting/alerting-rules/create-grafana-managed-rule.md [hunk 3]
```diff
@@ -143,7 +143,7 @@ To do this, you need to make sure that your alert rule is in the right evaluatio
 
    If you are creating a new evaluation group, specify the interval for the group.
 
-   All rules within the same group are evaluated concurrently over the same time interval.
+   All rules within the same group are evaluated over the same time interval.
```
Comment (yuri-tceretian) [inline docs/sources/alerting/alerting-rules/create-grafana-managed-rule.md:146]: I think the term concurrently is still correct here. However, it's not a true concurrency. It is correct that rules in the same group will all be evaluated against the same time stamp X. However, in absolute terms, they are evaluated at different times. 

In normal conditions, the scheduler ticks every 10s, e.g. at 9:30:00, 9:30:10, 9:30:20 etc. Every tick determines what rule group should be evaluated (using hashing to spread the evaluation of groups with the same eval interval).
After determining what groups should be evaluated, it takes all the rules in those groups and schedules their evaluation for that tick. However, the evaluation for that tick does not start immediately, scheduler delays the evaluation of each rule for up to 10 seconds. It evenly spreads the delay. For example, at tick T there are 100 rules to evaluate, therefore the first rule will start evaluating immediately, the second after 100ms, the third - after 200ms etc. 
It is worth mentioning that all those 100 rules will use the tick's timestamp to query data source, so it creates an illusion that they are all evaluated at the same time.

**Review by brendamuir — APPROVED** [review]

File: docs/sources/alerting/alerting-rules/create-grafana-managed-rule.md (hunk 3 above)
Comment (ppcano) [reply docs/sources/alerting/alerting-rules/create-grafana-managed-rule.md:146]: @yuri-tceretian, thanks for the detailed explanation.

I have updated a bit the Alert rule evaluation / Evaluation group as follows:

> **Grafana-managed** alert rules within the same group are evaluated at different times over the same evaluation interval but display the same evaluation timestamp.

Would this work?

I’m not sure if it’s necessary to document the behavior to this level of detail. I’m happy to help if we want to include this, but I think it should be outside of the Introduction/fundamentals section.

**Review by brendamuir — APPROVED** [review]

File: docs/sources/alerting/alerting-rules/create-grafana-managed-rule.md (hunk 3 above)
Comment (yuri-tceretian) [reply docs/sources/alerting/alerting-rules/create-grafana-managed-rule.md:146]: I would keep `concurrent` even without digging into details.

Comment (ppcano) [reply docs/sources/alerting/alerting-rules/create-grafana-managed-rule.md]: Addressed by https://github.com/grafana/grafana/pull/88250/commits/a3a0a43f141a82c6ba000004072b771a7086e50e and https://github.com/grafana/grafana/pull/88250/commits/96039b3d7ee3ad72922f4f9b78f0aeecd5ffa35d.  Keep `concurrent` and format the details for clarity.

**Review by yuri-tceretian — APPROVED** [review]

**Pull request merged**